### ISSUE TITLE
[babel 8] Enforce specifying proper `core-js` version

### ISSUE THREAD

**Issue by zloirock** [babel/babel#16042]
In most cases of `core-js` usage with `preset-env`, I see something like `corejs: 3` in the config. With installed `core-js@3.33`.

The first `core-js@3` release was in March 2019 and since then the number of `core-js` features has almost doubled.

`corejs: 3` in the `preset-env` config is interpreted `core-js@3.0` which limits the subset of used `core-js` features only to features that were in `core-js@3.0` and, in most cases, is not that the user expect.

I propose to limit potentially incorrect `corejs` option values.

- Disallow SemVer strings without a minor component, instead of `corejs: "3"` it should be at least `corejs: "3.0"`
- Disallow numbers: even if someone writes `corejs: 3.30` it will be interpreted as `corejs: 3.3`

---

From @JLHwung:

What about we also offer two magic options in Babel 8?

- `corejs: "node_modules"`: use whatever version in `./node_modules/core-js/package.json`
    This is for the people who bundle `core-js` with their website
- `corejs: "package.json"`: use the _minimum_ version specified in `packageJson.dependencies["core-js"]`
    This is for library authors that specified `core-js` as a dependency
 
For more sophisticated usage, please specify your own `core-js` version.

**Comment by nicolo-ribaudo** [comment]
Strong support for this. While removing support for core-js 2 I was thinking "maybe we should change the default to 3.33 instead of 3.0", but this idea is better.

Do you think we should allow semver strings with a _patch_ component?

**Comment by zloirock** [comment]
> maybe we should change the default to 3.33 instead of 3.0

The problem is that for any reason someone could have an older `core-js` version. And I hope on `core-js@4` release soon -)

> Do you think we should allow semver strings with a patch component?

By SemVer, new features are added only in minor releases, but if it's a valid version - why not? I think that it should be allowed.

**Comment by nicolo-ribaudo** [comment]
This should also be done in https://github.com/babel/babel-polyfills/tree/main/packages/babel-plugin-polyfill-corejs3.

**Comment by JLHwung** [comment]
What about we also offer two magic options in Babel 8?

- `corejs: "node_modules"`: use whatever version in `./node_modules/core-js/package.json`
    This is for the people who bundle `core-js` with their website
- `corejs: "package.json"`: use the _minimum_ version specified in `packageJson.dependencies["core-js"]`
    This is for library authors that specified `core-js` as a dependency
 
For more sophisticated usage, please specify your own `core-js` version.

**Comment by zloirock** [comment]
@JLHwung yep, I wanted to propose something like that, but was not sure about options names. Your option looks great -)

**Comment by zloirock** [comment]
Let's reopen it to implementation of @JLHwung proposal.
